### PR TITLE
[APB-4094][MP] Convert deprecated ContinueUrl to RedirectUrl

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/RedirectUrlActions.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import uk.gov.hmrc.agentsubscriptionfrontend.connectors.SsoConnector
+import uk.gov.hmrc.agentsubscriptionfrontend.service.SessionStoreService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl._
+import uk.gov.hmrc.play.bootstrap.binders.{AbsoluteWithHostnameFromWhitelist, RedirectUrl, UnsafePermitAll}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.higherKinds
+import scala.util.{Failure, Success, Try}
+
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/RedirectUrlActions.scala]: Hey, I know comments are seen as an anti-pattern, but it would be really nice to have a simple comment block at the top of this file which describes exactly what a "RedirectUrl" is in this context.  People might think it's just a standard HTTP 303, but I think the intention here is something different?  This is regarding what happens at the end of a journey, right?  Since RedirectUrl is such a general term, I think a comment would be helpful :)

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/RedirectUrlActions.scala]: 👍

File: conf/application.conf [hunk 2]
```diff
@@ -123,7 +123,6 @@ Dev {
       sso {
         host = localhost
         port = 9974
-        protocol = https
```
Comment (stevenmallen) [inline conf/application.conf]: Why was this removed?

File: it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala [hunk 3]
```diff
@@ -51,17 +51,17 @@ class SignOutControllerISpec extends BaseISpec {
         s"continue=%2Fagent-subscription%2Freturn-after-gg-creds-created%3Fid%3Dfoo")
     }
 
-    def assertContinueUrl(result: Result, continueUrl: ContinueUrl): Assertion = {
+    def assertContinueUrl(result: Result, continueUrl: String): Assertion = {
       val sosContinueValueUnencoded =
-        s"/agent-subscription/return-after-gg-creds-created?continue=${continueUrl.encodedUrl}"
+        s"/agent-subscription/return-after-gg-creds-created?continue=${URLEncoder.encode(continueUrl, "UTF-8")}"
```
Comment (stevenmallen) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala]: Thank you for spotting and fixing this :)

File: conf/application.conf (hunk 2 above)
Comment (mariannepearson0) [reply conf/application.conf]: when running it locally I was getting a no SSL certificate error (or something like this) when this endpoint was called because It should be http protocol in local...? http is the default if no protocol is provided
